app/notification_push.go

```diff
@@ -53,19 +54,26 @@ func (hub *PushNotificationsHub) GetGoChannelFromUserId(userId string) chan Push
 
 func (a *App) sendPushNotificationSync(post *model.Post, user *model.User, channel *model.Channel, channelName string, senderName string,
 	explicitMention bool, channelWideMention bool, replyToThreadType string) *model.AppError {
+	msg := a.BuildPushNotificationMessage(post, user, channel, channelName, senderName, explicitMention, channelWideMention, replyToThreadType)
+
+	return a.sendPushNotificationToAllSessions(msg, user.Id, "")
+}
 
-	sessions, err := a.getMobileAppSessions(user.Id)
+func (a *App) sendPushNotificationToAllSessions(msg model.PushNotification, userId string, skipSessionId string) *model.AppError {
+	sessions, err := a.getMobileAppSessions(userId)
 	if err != nil {
 		return err
 	}
 
-	msg := a.BuildPushNotificationMessage(post, user, channel, channelName, senderName, explicitMention, channelWideMention, replyToThreadType)
-
 	for _, session := range sessions {
 		if session.IsExpired() {
 			continue
 		}
 
+		if skipSessionId != "" && skipSessionId == session.Id {
```

Comment:
can `session.Id` be blank? i'd just check for being the same

we can probably join this with the previous check as there is no different action.
```
if session.IsExpired() || skipSessionId == session.Id {
    continue
}
```

Reply:
I don't think we actually have anything preventing the session ID from being blank, but it shouldn't ever be blank in practice. Like other IDs, it should always be a 26 character hex value.

Yeah, I can combine those lines. If the conditions to skip a session were more complicated, I might keep them separate to clarify the different reasons we're skipping that loop, but these ones are simple enough